hack/list-images.sh

```diff
@@ -0,0 +1,135 @@
+
+# Dummy variables
+declare -rx releaseNamespace=unused \
+            RES="" \
+            targetNamespace=unused \
+            generatedName=unused \
+            name=unused \
+            namespace=unused \
+            clusterID=unused \
+            prometheusService=unused \
+            releaseName=unused \
+            valuesFrom=unused \
+            workspaceNamespace=unused \
+            chartmuseumAdminCredentialsSecret=unused \
+            certificateIssuerName=unused \
+            verb=unused \
+            objectRef=unused \
+            user=unused \
+            adminCredentialsSecret=unused \
+            tlsCertificateSecret=unused \
+            airgappedEnabled=true \
+            kommanderAuthorizedlisterImageTag='' \
+            kommanderAuthorizedlisterImageRepository='' \
+            certificatesCAIssuerName=unused \
+            caSecretNamespace=unused \
+            kommanderControllerManagerImageTag='' \
+            kommanderControllerManagerImageRepository='' \
+            kommanderFluxNamespace=unused \
+            kommanderGitCredentialsSecretName=unused \
+            ageEncryptionSecretName=unused \
+            ageEncryptionSecretKey=unused \
+            kommanderControllerWebhookImageTag='' \
+            kommanderControllerWebhookImageRepository='' \
+            kommanderFluxOperatorManagerImageTag='' \
+            kommanderFluxOperatorManagerImageRepository='' \
+            certificatesIssuerName=unused \
+            caSecretName=unused \
+            kommanderLicensingControllerManagerImageTag='' \
+            kommanderLicensingControllerManagerImageRepository='' \
+            kommanderLicensingControllerWebhookImageTag='' \
+            kommanderLicensingControllerWebhookImageRepository='' \
+            kommanderAppManagementReplicas='' \
+            kommanderAppManagementImageTag='' \
+            kommanderAppManagementImageRepository='' \
+            kommanderAppManagementKubetoolsImageRepository='' \
+            kommanderAppManagementWebhookImageRepository='' \
+            kommanderAppManagementConfigAPIImageRepository='' \
+            tfaName=unused \
+            notPopulatedAnywhereAsThisIsOnlyForAirgappedBundle=unused \
+            caIssuerName=unused \
+            kommanderChartVersion="${kommanderChartVersion:-}"
+
+IMAGES_FILE="$(realpath "$(mktemp .helm-list-images-XXXXXX)")"
+readonly IMAGES_FILE
+trap_add "rm --force ${IMAGES_FILE}" EXIT
+
+for dir in $(find . -type f -name "*.yaml" -print0 | xargs --null --max-lines=1 --no-run-if-empty -- grep --files-with-matches '^kind: HelmRelease' | grep --only-matching "\(.*\)/" | sort --unique); do
+  pushd "${dir}" &>/dev/null
+
+  while IFS= read -r hr; do
+    pushd "$(dirname "${hr}")" &>/dev/null
+    extra_args=()
+    defaults_cm=""
+    if [ -f 'defaults/cm.yaml' ]; then
+      defaults_cm='defaults/cm.yaml'
+    elif [ -f '../defaults/cm.yaml' ]; then
+      defaults_cm='../defaults/cm.yaml'
+    fi
```

Comment:
This looks like we need an additional CI check to make sure that default values are not stored in any other files and use this pattern only.

Reply:
This has been the convention so far but yes might be good to add another check.

Reply:
Can we do that as a follow up though? :sweat_smile:

Reply:
Definitely.